internal/releasesources/release_sources.go

```diff
@@ -0,0 +1,528 @@
+	if err != nil {
+		return err
+	}
+
+	dynamicEnrichedReleaseImages, err = h.removeReleaseImageDuplicates(dynamicEnrichedReleaseImages)
+	if err != nil {
+		return err
+	}
+
+	EnrichedReleaseImages := h.mergeEnrichedReleaseImages(staticEnrichedReleaseImages, dynamicEnrichedReleaseImages)
+
+	h.log.Debug("Starting SQL transaction")
+	tx := h.db.Begin()
+	// Deleting all releases before adding again in order to
+	// store only the releases according to RELEASE_SOURCES
+	err = h.deleteAllReleases(tx)
+	if err != nil {
+		tx.Rollback()
+		return err
+	}
+
+	err = h.insertToDB(tx, EnrichedReleaseImages)
+	if err != nil {
+		tx.Rollback()
+		return err
+	}
+
+	h.log.Debug("Commiting changes")
+	return tx.Commit().Error
+}
+
+// getReleaseImagesMajorOCPVersions retrieves the major OCP versions for both static and dynamic release images discovered.
+func (h *releaseSourcesHandler) getReleaseImagesMajorOCPVersions() (ocpMajorVersionSet, error) {
+	majorVersions := ocpMajorVersionSet{}
+
+	// First from static release images
+	for _, releaseImage := range h.staticReleaseImages {
+		majorVersion, err := common.GetMajorVersion(*releaseImage.OpenshiftVersion)
+		if err != nil {
+			return nil, errors.Errorf("error occurred while trying to get the major version of %s", *releaseImage.OpenshiftVersion)
+		}
+		majorVersions[*majorVersion] = true
+	}
+
+	// Then from dynamic release images
+	for _, releaseSource := range h.releaseSources {
+		majorVersion, err := common.GetMajorVersion(*releaseSource.OpenshiftVersion)
+		if err != nil {
+			return nil, errors.Errorf("error occurred while trying to get the major version of %s", *releaseSource.OpenshiftVersion)
+		}
+		majorVersions[*majorVersion] = true
+	}
+
+	return majorVersions, nil
+}
+
+// getSupportLevels retrieves a mapping from OCP major.minor versions
+// to their corresponding support levels for both static and dynamic release images.
+func (h *releaseSourcesHandler) getSupportLevels() (ocpVersionSupportLevels, error) {
+	ocpMajorVersionSet, err := h.getReleaseImagesMajorOCPVersions()
+	if err != nil {
+		return nil, err
+	}
+
+	supportLevels := ocpVersionSupportLevels{}
+
+	for majorVersion := range ocpMajorVersionSet {
+		supportLevelsForMajor, err := h.supportLevelClient.getSupportLevels(majorVersion)
+		if err != nil {
+			return nil, err
+		}
+
+		for majorMinorVersion, supportLevel := range supportLevelsForMajor {
+			supportLevels[majorMinorVersion] = supportLevel
+		}
+	}
+
+	return supportLevels, nil
+}
+
+// setSupportLevels sets support level for given release images as follows:
+// candidate/pre-release releases are considered support level 'beta'.
+// The rest are stable releases, and their support level is set according to the mapping
+// retrieved from the API.
+func (h *releaseSourcesHandler) setSupportLevels(releases []*enrichedReleaseImage, supportLevels ocpVersionSupportLevels) error {
+	for _, release := range releases {
+		if release.SupportLevel != "" {
+			continue
+		}
```

Comment:
This will need to be removed if we persist enrichedReleaseImages for static.

Currently this will always we empty for static release images, but if we save them it will always have the value from the previous loop. To deal with this we can just always set the level to the new one for this loop.